README.md

```diff
@@ -1,29 +1,78 @@
 
-diagram that shows the mirroring of CRUD, to ROutes to controller Actions
+If you were to build out a feature in Rails that entails all of the potential RESTful routes, you would have seven different RESTful URLs, HTTP verbs (more on those later), and actions. As an example, imagine that you have a Newsletter module in your application, if you run ```rake routes``` in the terminal you will receive the following output:
+
+```
+newsletters     GET     /newsletters(.:format)          newsletters#index
+                POST    /newsletters(.:format)          newsletters#create
+new_newsletter  GET     /newsletters/new(.:format)      newsletters#new
+edit_newsletter GET     /newsletters/:id/edit(.:format) newsletters#edit
+newsletter      GET     /newsletters/:id(.:format)      newsletters#show
+                PATCH   /newsletters/:id(.:format)      newsletters#update
+                PUT     /newsletters/:id(.:format)      newsletters#update
+                DELETE  /newsletters/:id(.:format)      newsletters#destroy
+```
+
+Rails does a great job of integrating RESTful routes into their system, if you can understand routes in Rails you can understand REST in general. Going through the output above you should be able to see that all of the potential CRUD actions are all there, from querying all of the records to deleting a single item from the database, and all of the actions are wired up using RESTful routing nomenclature.
+
+Here is a diagram that shows how how the views, controller actions, routes, and HTTP verbs are all mapped together
+
+![REST Diagram](http://reif.io/lib/flatiron/rest_diagram.png)
+
+In analyzing the diagram, you can see the flow of data as follows:
+
+1. The HTTP request contains a HTTP verb and hits a specific URL path
+
+2. The router in the application processes the request and 'routes' the request data to the proper controller action
+
+3. The controller action either performs a task, such as creating, updating, or deleting a record in the database, or it runs a database query and renders a view to the client
+
+## Definition of HTTP Verbs
+
+So what do those 'GET', 'POST', et al items represent? Those are HTTP verbs that each give the HTTP request unique behavior, below is an explanation on each verb:
+
+* **GET** - The GET method means retrieve whatever information (in the form of an entity) is identified by the Request-URI. This means if you go to /posts you will get all of the posts that the application wants your application to have.
+
+* **POST** - The POST method is used to request that the origin server accept the entity enclosed in the request as a new subordinate of the resource identified by the Request-URI in the Request-Line. This means that the server needs to accept the data sent over with the request.
```

Comment:
This sentence is super jargon-y: Maybe something like `The POST method is used to send data enclosed in the request to the server. The server is expected to use this data to create some new resource`